pact/consumer/core/src/main/java/org/arquillian/algeron/pact/consumer/core/client/PactConsumerArchiveAppender.java

```diff
@@ -56,6 +68,61 @@
         return merge;
     }
 
+    private JavaArchive addSunHttpServer(JavaArchive arquillianPactConsumer) {
+        try {
+            arquillianPactConsumer.addClass(Class.forName("com.sun.net.httpserver.spi.HttpServerProvider$1"))
```

Comment:
I think I know what might be wrong with the jar. I encountered similar thing before.

  * starting point: https://issues.jboss.org/browse/SHRINKWRAP-446
  * reproducer:  https://github.com/bartoszmajsak/groovy-classloading
  * related groovy issue: https://issues.apache.org/jira/browse/GROOVY-6158

Let's see if that's the same problem in this case.

If so, there is not so much we can do about it at this point, as those classes are coming from gazzilion possible JDK versions :)

We can however simplify this code I believe:
  * using Shrinkwrap we can probably find all the classes by name matching a pattern, right?
  * we can then return a array of `Class` instances created in the same way as now - `.forName`
  * use `.addClasses(...)` on the enriched archive